### DISCUSSION TITLE
How do action redirects work?

### DISCUSSION THREAD

**Discussion by kristojorg** [remix-run/remix#1799] [Q&A] [answered]
When I create a form with a submit button and use `Form`, my understanding is that it does not reload the page, but uses the `onSubmit` to send a request the the endpoint listed in `action="/endpoint`. So when an action returns a redirect, how does that work? Does the page actually get redirected? If so, how? Is there some logic in the `onSubmit` to detect a redirect response and redirect the page there? Would that work without javascript?

**Comment by sergiodxa** [comment, marked as answer]
The Form component receives the response with the redirect from the server and does a client-side navigation, unless you used the reloadDocument prop on the Form component.

If JS is disabled or not loaded, the Form will behave as a normal form tag and it will do a full page navigation which then will be redirected, in this case the browser handles everything.

**Reply by kristojorg** [reply]
Ok nice, good to know thank you!
